File: src/test/java/org/oewntk/xml/in/TestsXmlModelWords.java
/*
 * Copyright (c) 2021-2021. Bernard Bou.
 */

package org.oewntk.xml.in;

import org.junit.BeforeClass;
import org.junit.Test;
import org.oewntk.model.Key;
import org.oewntk.model.LibTestModel;
import org.xml.sax.SAXException;

import java.io.IOException;
import java.util.Set;

import javax.xml.parsers.ParserConfigurationException;

public class TestsXmlModelWords
{
	private static final boolean peekTestWords = false;

	private static final Set<String> testWords = Set.of("baroque", "Baroque", "bass", "row");

	@BeforeClass
	public static void init() throws IOException, ParserConfigurationException, SAXException
	{
		TestsXmlCommon.init();
	}

	@Test
	public void testScanLexesForTestWords()
	{
		LibTestModel.testScanLexesForTestWords(TestsXmlCommon.model, Key.W_P_A::of_t, LibTestModel::makeIndexMap, testWords, peekTestWords, TestsXmlCommon.ps);
	}

	@Test
	public void testScanLexesForTestWordsSorted()
	{
		LibTestModel.testScanLexesForTestWords(TestsXmlCommon.model, Key.W_P_A::of_t, LibTestModel::makeSortedIndexMap, testWords, peekTestWords, TestsXmlCommon.ps);
	}

	@Test
	public void testBass()
	{
		LibTestModel.testWord("bass", TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testRow()
	{
		LibTestModel.testWord("row", TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testBaroque()
	{
		LibTestModel.testWords(TestsXmlCommon.model, TestsXmlCommon.ps, "baroque", "Baroque");
	}

	@Test
	public void testEarth()
	{
		LibTestModel.testWords(TestsXmlCommon.model, TestsXmlCommon.ps, "earth", "Earth");
	}

	@Test
	public void testCritical()
	{
		LibTestModel.testWord("critical", 'a', TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testHollywood()
	{
		LibTestModel.testWord("Hollywood", 'a', TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testVictorian()
	{
		LibTestModel.testWord("Victorian", 'a', TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testAllied()
	{
		LibTestModel.testWord("allied", 'a', TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testAlliedUpper()
	{
		LibTestModel.testWord("Allied", 'a', TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testAbsent()
	{
		LibTestModel.testWord("absent", 'a', TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testApocryphal()
	{
		LibTestModel.testWord("apocryphal", 'a', TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testUsed()
	{
		LibTestModel.testWord("used", 'a', TestsXmlCommon.model, TestsXmlCommon.ps);
	}
}

File: src/test/java/org/oewntk/xml/in/TestsXmlModelWords.kt
/*
 * Copyright (c) 2021-2024. Bernard Bou.
 */
package org.oewntk.xml.`in`

import org.junit.BeforeClass
import org.junit.Test
import org.oewntk.model.Key
import org.oewntk.model.Lemma
import org.oewntk.model.Lex
import org.oewntk.model.LibTestModel.makeIndexMap
import org.oewntk.model.LibTestModel.makeSortedIndexMap
import org.oewntk.model.LibTestModel.testScanLexesForTestWords
import org.oewntk.model.LibTestModel.testWord
import org.oewntk.model.LibTestModel.testWords
import org.oewntk.model.PartOfSpeech
import org.oewntk.xml.`in`.LibTestsXmlCommon.model
import org.oewntk.xml.`in`.LibTestsXmlCommon.ps

class TestsXmlModelWords {

    @Test
    fun testScanLexesForTestWords() {
        testScanLexesForTestWords(model, { lex: Lex -> Key.UsingPronunciation.of(lex) }, { seq: Sequence<Key> -> makeIndexMap(seq) }, testWords, PRINT_TEST_WORDS, ps)
    }

    @Test
    fun testScanLexesForTestWordsSorted() {
        testScanLexesForTestWords(model, { lex: Lex -> Key.UsingPronunciation.of(lex) }, { seq: Sequence<Key> -> makeSortedIndexMap(seq) }, testWords, PRINT_TEST_WORDS, ps)
    }

    @Test
    fun testBass() {
        testWord(Lemma("bass"), model, ps)
    }

    @Test
    fun testRow() {
        testWord(Lemma("row"), model, ps)
    }

    @Test
    fun testBaroque() {
        testWords(model, ps, arrayOf(Lemma("baroque"), Lemma("Baroque")))
    }

    @Test
    fun testEarth() {
        testWords(model, ps, arrayOf(Lemma("earth"), Lemma("Earth")))
    }

    @Test
    fun testCritical() {
        testWord(Lemma("critical"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testHollywood() {
        testWord(Lemma("Hollywood"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testVictorian() {
        testWord(Lemma("Victorian"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testAllied() {
        testWord(Lemma("allied"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testAlliedUpper() {
        testWord(Lemma("Allied"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testAbsent() {
        testWord(Lemma("absent"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testApocryphal() {
        testWord(Lemma("apocryphal"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testUsed() {
        testWord(Lemma("used"), PartOfSpeech.A, model, ps)
    }

    companion object {

        private const val PRINT_TEST_WORDS = false

        private val testWords = setOf(Lemma("baroque"), Lemma("Baroque"), Lemma("bass"), Lemma("row"))

        @JvmStatic
        @BeforeClass
        fun init() {
            model
        }
    }
}
